module/applications/activity/activity-sheet.mjs

```diff
@@ -0,0 +1,494 @@
+        id: "identity", group: "sheet", icon: "fa-solid fa-tag",
+        label: "DND5E.ACTIVITY.SECTIONS.Identity"
+      },
+      activation: {
+        id: "activation", group: "sheet", icon: "fa-solid fa-clapperboard",
+        label: "DND5E.ACTIVITY.SECTIONS.Activation",
+        tabs: {
+          time: {
+            id: "time", group: "activation", icon: "fa-solid fa-clock",
+            label: "DND5E.ACTIVITY.SECTIONS.Time"
+          },
+          consumption: {
+            id: "consumption", group: "activation", icon: "fa-solid fa-boxes-stacked",
+            label: "DND5E.ACTIVITY.FIELDS.consumption.label"
+          },
+          targeting: {
+            id: "activation-targeting", group: "activation", icon: "fa-solid fa-bullseye",
+            label: "DND5E.ACTIVITY.FIELDS.target.label"
+          }
+        }
+      },
+      effect: {
+        id: "effect", group: "sheet", icon: "fa-solid fa-sun",
+        label: "DND5E.ACTIVITY.SECTIONS.Effect"
+      }
+    });
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Helper to mark the tabs data structure with the appropriate CSS class if it is active.
+   * @param {Record<string, Partial<ApplicationTab>>} tabs  Tabs definition to modify.
+   * @returns {Record<string, Partial<ApplicationTab>>}
+   * @internal
+   */
+  _markTabs(tabs) {
+    for ( const v of Object.values(tabs) ) {
+      v.active = this.tabGroups[v.group] === v.id;
+      v.cssClass = v.active ? "active" : "";
+      if ( "tabs" in v ) this._markTabs(v.tabs);
+    }
+    return tabs;
+  }
+
+  /* -------------------------------------------- */
+  /*  Life-Cycle Handlers                         */
+  /* -------------------------------------------- */
+
+  /** @override */
+  _canRender(options) {
+    if ( !this.isVisible ) throw new Error(game.i18n.format("SHEETS.DocumentSheetPrivate", {
+      type: game.i18n.localize("DND5E.ACTIVITY.Label")
+    }));
+  }
+
+  /* -------------------------------------------- */
+
+  /** @inheritDoc */
+  _onFirstRender(context, options) {
+    super._onFirstRender(context, options);
+    this.activity.constructor._registerApp(this.activity, this);
+  }
+
+  /* -------------------------------------------- */
+
+  /** @override */
+  _onClose(_options) {
+    this.activity.constructor._unregisterApp(this.activity, this);
+  }
+
+  /* -------------------------------------------- */
+  /*  Event Listeners and Handlers                */
+  /* -------------------------------------------- */
+
+  /**
+   * Handle adding a new entry to the consumption list.
+   * @this {ActivityConfig}
+   * @param {Event} event         Triggering click event.
+   * @param {HTMLElement} target  Button that was clicked.
+   */
+  static #addConsumption(event, target) {
+    const types = this.activity.validConsumptionTypes;
+    const existingTypes = new Set(this.activity.consumption.targets.map(t => t.type));
+    const filteredTypes = types.difference(existingTypes);
```

Comment:
Feels like this logic makes a bit less sense for consumption targets. I would have thought a pretty common use case would be to, for example, -1 charges from one item, and +1 charges to another item. I don't feel strongly though.